Ginger/GingerCoreCommonTest/VariableTests/RandomVariableTest.cs

```diff
@@ -0,0 +1,454 @@
+            
+            //Act
+            variableRandomString.GenerateAutoValue();
+
+            //Assert
+           Assert.AreEqual(variableRandomString.Value, variableRandomString.Value.ToUpper(), "variableRandomString.Value, variableRandomString.Value.ToUpper()");
+        }
+
+        [TestMethod]
+        public void RandomStringVar_BothupperlowerString15char_ShowError()
+        {
+            //Arrange
+            VariableRandomString variableRandomString = new VariableRandomString();
+            variableRandomString.Max = 15;
+            variableRandomString.IsUpperCase = true;            
+            
+            //Act
+            //When the user change to to lower case make sure the Isupper case get turned off
+            variableRandomString.IsLowerCase = true;
+            variableRandomString.GenerateAutoValue();
+
+            //Assert
+            Assert.AreEqual(variableRandomString.IsUpperCase, false, "variableRandomString.IsUpperCase=false");
+            Assert.AreEqual(variableRandomString.Value, variableRandomString.Value.ToLower());
+
+            //TODO: verify the formula
+            //Assert.AreEqual(variableRandomString.Formula, "Error", "variableRandomString.Formula=Error");
+        }
+
+        [TestMethod]
+        public void RandomStringVar_0_3_chars()
+        {
+            //Arrange
+            VariableRandomString variableRandomString = new VariableRandomString();
+            variableRandomString.Min = 0;
+            variableRandomString.Max = 10;
+            bool Hit0 = false;
+            bool Hit5 = false;
+            bool Hit10 = false;
+
+            //Act            
+            for (int i = 0; i < 100;i++ )
+            {
+                variableRandomString.GenerateAutoValue();
+                Assert.IsTrue(variableRandomString.Value.Length >= 0 && variableRandomString.Value.Length <= 10, "variableRandomString.Value.Length >= 0 && variableRandomString.Value.Length <= 3");                
+                if (variableRandomString.Value.Length == 0) Hit0 = true;
+                if (variableRandomString.Value.Length == 5) Hit5 = true;
+                if (variableRandomString.Value.Length == 10) Hit10 = true;                
+            }
+
+            //Verify we hit the boundries at least once
+            Assert.IsTrue(Hit0, "Hit0");
+            Assert.IsTrue(Hit5, "Hit5");
+            Assert.IsTrue(Hit10, "Hit10");
+
+        }
+
+        [TestMethod]
+        public void RandomStringVar_DigitsString15char()
+        {
+            //Arrange
+            VariableRandomString variableRandomString = new VariableRandomString();
+            variableRandomString.Max = 15;            
+            variableRandomString.IsDigit = true;
+            
+            //Act
+            variableRandomString.GenerateAutoValue();
+
+            //Assert
+           Assert.AreEqual(variableRandomString.Value, variableRandomString.Value.ToLower());
```

Comment:
incorrect - need to verify all chars are digit

Reply:
Fixed - modified the code to validate if the string is all having numbers